agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -201,6 +201,7 @@ def start(info)
         end
       end
 
+      attach_router unless interface_ip('weave')
```

Comment:
What happens if the weave interface exists, but it doesn't have any address configured?

Looking at  #1165 and `Errno::EADDRNOTAVAIL`...